blocks/library/quote/index.js

```diff
@@ -30,21 +31,52 @@ registerBlock( 'core/quote', {
 		}
 	},
 
-	controls: [ 1, 2 ].map( ( variation ) => ( {
-		icon: 'format-quote',
-		title: wp.i18n.sprintf( wp.i18n.__( 'Quote style %d' ), variation ),
-		isActive: ( { style = 1 } ) => style === variation,
-		onClick( attributes, setAttributes ) {
-			setAttributes( { style: variation } );
-		},
-		subscript: variation
-	} ) ),
+	controls: [
+		[
+			{
+				icon: 'editor-alignleft',
+				title: wp.i18n.__( 'Align left' ),
+				isActive: ( { align } ) => ! align || 'left' === align,
+				onClick( attributes, setAttributes ) {
+					setAttributes( { align: undefined } );
+				}
+			},
+			{
+				icon: 'editor-aligncenter',
+				title: wp.i18n.__( 'Align center' ),
+				isActive: ( { align } ) => 'center' === align,
+				onClick( attributes, setAttributes ) {
+					setAttributes( { align: 'center' } );
+				}
+			},
+			{
+				icon: 'editor-alignright',
+				title: wp.i18n.__( 'Align right' ),
+				isActive: ( { align } ) => 'right' === align,
+				onClick( attributes, setAttributes ) {
+					setAttributes( { align: 'right' } );
+				}
+			}
+		],
+		[ 1, 2 ].map( ( variation ) => ( {
+			icon: 'format-quote',
+			title: wp.i18n.sprintf( wp.i18n.__( 'Quote style %d' ), variation ),
+			isActive: ( { style = 1 } ) => style === variation,
+			onClick( attributes, setAttributes ) {
+				setAttributes( { style: variation } );
+			},
+			subscript: variation
+		} ) )
+	],
 
 	edit( { attributes, setAttributes, focus, setFocus } ) {
-		const { value, citation, style = 1 } = attributes;
+		const { align, value, citation, style = 1 } = attributes;
 
 		return (
-			<blockquote className={ `blocks-quote blocks-quote-style-${ style }` }>
+			<blockquote
+				className={ `blocks-quote blocks-quote-style-${ style }` }
+				style={ align ? { textAlign: align } : null }
```

Comment:
Sorry for the midstream correction on this one. Here's a mockup of a multi paragraph quote:

![quote left aligned](https://cloud.githubusercontent.com/assets/1204802/25490031/ed65396c-2b6b-11e7-9994-09865282b6e8.png)

Note how the caret is in text that is left aligned, therefore the toolbar shows the left alignment highlighted.

Now the first paragraph is right aligned:

![quote right aligned](https://cloud.githubusercontent.com/assets/1204802/25490047/fdead04e-2b6b-11e7-8155-2ae902f84c4d.png)

Note how the caret is inside the first paragraph.

Technically the behavior here would be the same as that of the "Continuous text" block.